--- src/TerrariaNetCore/Terraria/GameContent/UI/Elements/UIKeybindingListItem.cs
+++ src/tModLoader/Terraria/GameContent/UI/Elements/UIKeybindingListItem.cs
@@ -4,6 +_,7 @@
 using Terraria.GameContent.UI.Chat;
 using Terraria.GameInput;
 using Terraria.Localization;
+using Terraria.ModLoader;
 using Terraria.UI;
 using Terraria.UI.Chat;
 
@@ -105,6 +_,11 @@
 		return _keybind switch {
 			"MouseLeft" => Lang.menu[162].Value,
 			"MouseRight" => Lang.menu[163].Value,
+			// Support for extra mouse buttons: [[
+			"MouseMiddle" => Language.GetTextValue("tModLoader.MouseMiddle"),
+			"MouseXButton1" =>Language.GetTextValue("tModLoader.MouseXButton1"),
+			"MouseXButton2" => Language.GetTextValue("tModLoader.MouseXButton2"),
+			// ]]
 			"Up" => Lang.menu[148].Value,
 			"Down" => Lang.menu[149].Value,
 			"Left" => Lang.menu[150].Value,
@@ -155,6 +_,8 @@
 			"Loadout2" => Language.GetTextValue("UI.Loadout2"),
 			"Loadout3" => Language.GetTextValue("UI.Loadout3"),
 			"ToggleCameraMode" => Language.GetTextValue("UI.ToggleCameraMode"),
+			// Handle mod keybinds
+			_ when KeybindLoader.modKeybinds.TryGetValue(_keybind, out var modKeybind) => modKeybind.DisplayName.GetTranslation(Language.ActiveCulture).
 			_ => _keybind,
 		};
 	}
