--- src/TerrariaNetCore/Terraria/GameInput/PlayerInputProfile.cs
+++ src/tModLoader/Terraria/GameInput/PlayerInputProfile.cs
@@ -1,12 +_,14 @@
 using System;
 using System.Collections.Generic;
+using System.Linq;
 using Microsoft.Xna.Framework.Input;
 using Newtonsoft.Json;
 using Newtonsoft.Json.Linq;
+using Terraria.ModLoader;
 
 namespace Terraria.GameInput;
 
-public class PlayerInputProfile
+public partial class PlayerInputProfile
 {
 	public Dictionary<InputMode, KeyConfiguration> InputModes = new Dictionary<InputMode, KeyConfiguration> {
 		{ InputMode.Keyboard, new KeyConfiguration() },
@@ -200,7 +_,8 @@
 	public void CopyGameplaySettingsFrom(PlayerInputProfile profile, InputMode mode)
 	{
 		string[] keysToCopy = new string[23] {
-			"MouseLeft", "MouseRight", "Up", "Down", "Left", "Right", "Jump", "Grapple", "SmartSelect", "SmartCursor",
+			// Patch to be improved after decompiler reconfiguration.
+			"MouseLeft", "MouseRight", TriggerNames.MouseMiddle, TriggerNames.MouseXButton1, TriggerNames.MouseXButton2, "Up", "Down", "Left", "Right", "Jump", "Grapple", "SmartSelect", "SmartCursor",
 			"QuickMount", "QuickHeal", "QuickMana", "QuickBuff", "Throw", "Inventory", "ViewZoomIn", "ViewZoomOut", "Loadout1", "Loadout2",
 			"Loadout3", "ToggleCreativeMenu", "ToggleCameraMode"
 		};
